pulsar-broker/src/main/java/org/apache/pulsar/broker/MessagingServiceShutdownHook.java

```diff
@@ -52,30 +49,15 @@ public void run() {
                     + service.getSafeWebServiceAddress() + ", broker url=" + service.getSafeBrokerServiceUrl());
         }
 
-        ExecutorService executor = Executors.newSingleThreadExecutor(new DefaultThreadFactory("shutdown-thread"));
-
         try {
-            CompletableFuture<Void> future = new CompletableFuture<>();
-
-            executor.execute(() -> {
-                try {
-                    service.closeAsync().whenComplete((result, throwable) -> {
-                        if (throwable != null) {
-                            future.completeExceptionally(throwable);
-                        } else {
-                            future.complete(result);
-                        }
-                    });
-                } catch (Exception e) {
-                    future.completeExceptionally(e);
-                }
-            });
-
-            future.get(service.getConfiguration().getBrokerShutdownTimeoutMs(), TimeUnit.MILLISECONDS);
-
+            service.closeAsync().get();
```

Comment:
Why not use the "shutdown-thread" here?  And if we don't apply the future timeout, does the `processTerminator.accept(0)` can works? The thread might always block by the get method.

Reply:
I reverted the changes so it's back to the original logic. The reason I had made the changes is the fact that BrokerService now contains timeout handling as well. However this code calls the shutdown of PulsarService so there could be possibly some call that blocks and the original logic is safer.